pkg/validation/options.go

```diff
@@ -187,6 +187,14 @@ func Validate(o *options.Options) error {
 		}
 	}
 
+	headerSplitRegex := regexp.MustCompile(`^([^:]*)=[ \t]*(.*)$`)
```

Comment:
I don't think we should allow `[ \t]*` after the `=` -- I think in other `key=value` parameters we force a no spaces syntax.

This obviously becomes much cleaner with structured configs in the future.

Reply:
You've also implemented this Regex in two areas -- can you get this centralized in 1 spot?

Reply:
And one last thing 😄 

Should `[^:]` in the key section be `[^=]` since we switched the `key=value` separator from `key:value`?

If we are intending this regex to do more besides internal consistency (IE legal header names and values) -- then we need to go beyond `[^:]` and `.*` for more specificity.

Reply:
> You've also implemented this Regex in two areas -- can you get this centralized in 1 spot?

Ok, we've just crossed over the "I'm not skilled enough at golang at this point to do this without significant amounts of time to learn how to use it properly" line.

Things are in different packages and its not clear where they belong either.

Could someone else please carry this over the finish line if they want that level of code quality? Alternately, could someone please tell me exactly how to do what you want done?

Thanks!

Reply:
Maybe for now just put the string part you are using to compile the regex in a central place as a public const.  How about here: https://github.com/oauth2-proxy/oauth2-proxy/blob/master/pkg/validation/options.go

`const ExtraHeaderSplitRegex = ...` then you can get at it with `validation.ExtraHeaderSplitRegex`.  We can sort out the long term organization as the validation package vs initialization logic responsibility split we are working matures.

Reply:
I also think this is the regex we want: `^([^=]+)=(.+)$` based on my notes above, WDYT?

Reply:
Ok, I think I managed to do that.